src/segmented-control/segmented-control.scss

```diff
@@ -0,0 +1,173 @@
+  // stylelint-disable-next-line scss/selector-no-redundant-nesting-selector
+  & + .SegmentedControl-button::before {
+    position: absolute;
+    top: 1px; // Account for border width
+    left: -1px;
+    height: var(--primer-text-body-size-large, 16px);
+    // stylelint-disable-next-line primer/spacing
+    margin-top: var(--primer-control-medium-paddingBlock, 6px);
+    content: '';
+    // stylelint-disable-next-line primer/borders
+    border-left: var(--primer-borderWidth-thin, $border-width) $border-style var(--color-border-default);
+    transition: border-color 180ms cubic-bezier(0.3, 0.1, 0.5, 1);
+  }
+
+  // Remove dividers
+
+  &[aria-pressed='true'] + .SegmentedControl-button::before {
+    border-color: transparent;
+  }
+
+  &:hover,
+  &:active,
+  &:focus-visible {
+    &::before,
+    + .SegmentedControl-button::before {
+      border-color: transparent;
+      transition-duration: 60ms;
+    }
+  }
+}
+
+// Icon -----------------------------------------
+
+.SegmentedControl-leadingIcon {
+  color: var(--color-fg-muted);
+}
+
+// Label -----------------------------------------
+
+.SegmentedControl-label {
+  // renders a visibly hidden "copy" of the label in bold, reserving box space for when label becomes bold on selected
+  &[data-content]::before {
+    display: block;
+    height: 0;
+    // stylelint-disable-next-line primer/typography
+    font-weight: var(--base-text-weight-semibold, $font-weight-bold);
+    visibility: hidden;
+    content: attr(data-content);
+  }
+}
+
+// Variants -----------------------------------------
+
+// disabled
+.SegmentedControl--disabled { // TODO: Replace with aria-disabled?
+  .SegmentedControl-button {
+    color: var(--color-primer-fg-disabled);
+    pointer-events: none;
+    cursor: default;
+
+    &[aria-pressed='true'] {
+      border-color: var(--color-border-default);
+    }
+  }
+
+  .SegmentedControl-leadingIcon {
+    color: var(--color-primer-fg-disabled);
+  }
+}
+
+.SegmentedControl-button--isLoading.SegmentedControl-button--isLoading {
+  pointer-events: none;
+  cursor: default;
+
+  .SegmentedControl-label,
+  .SegmentedControl-leadingIcon {
+    color: var(--color-primer-fg-disabled);
+  }
+}
+
+// fullWidth
+.SegmentedControl--fullWidth {
+  display: flex;
+
+  .SegmentedControl-button {
+    flex: 1;
+  }
+}
+
+// Icon only
+.SegmentedControl-button--iconOnly {
+  width: var(--primer-control-medium-size, 32px);
+  padding-right: 0;
+  padding-left: 0;
+}
+
+// Icon only when narrow
+@media (max-width: $width-md) { // TODO: use token
+  .SegmentedControl--iconOnly-whenNarrow {
+    .SegmentedControl-button {
+      width: var(--primer-control-medium-size, 32px);
+      padding-right: 0;
+      padding-left: 0;
+    }
+
+    .SegmentedControl-label {
+      display: none;
+    }
+  }
+}
+
+// Increase touch target
+@media (pointer: coarse) {
+  .SegmentedControl-button::after {
+    @include minTouchTarget($min-height: var(--primer-control-minTarget-coarse, 44px));
```

Comment:
@simurai for the width to also pass min 44px of touch target, should we add a `min-width: 44px` equivalent, besides creating the pseudo-element for fixing the target's height?

Reply:
If the segment is <44px, a `min-width: 44px` equivalent would cause that segment to overlap with the adjacent segments' touch targets.

We could instead set `min-width: 44px` on the button itself.

Reply:
> We could instead set `min-width: 44px` on the button itself.

Yeah, I think we could do that except for the "icon only" variant where the segment should stay a square.